### DISCUSSION TITLE
Connecting to Fonoster using Mobile/WebRTC

### DISCUSSION THREAD

**Discussion by psanders** [fonoster/yaps#226] [Ideas]
Let's get this topic started :)

**Comment by itzmanish** [comment]
Hey, How do we achieve this?

**Reply by psanders** [reply]
Hi @itzmanish 

There are several use-cases within this topic. For the sake of starting this conversation, I will begin with simple calls initiated within the browser.

In a Fonoster Project, you can add VoIP Providers, Virtual Numbers, Domains, and Agents.

An Agent is a SIP endpoint that registers to Fonoster to make or receive calls. With an Agent, you can log in to Fonoster using a regular Softphone or the browser.

ALL Agents must belong to a Domain. Agents in a Domain can call each other, but they cannot call someone on another Domain. For calls from/to the mobile network, a Domain must have assigned a Virtual Number.

Virtual Numbers (or DID/DOD) must be obtained from a VoIP Provider and configured in Fonoster.

So…

If you want to make a call from the browser:

1. Find a VoIP Provider and obtain the SIP credentials
2. Buy a Virtual Number from your VoIP Provider
3. Add your VoIP Provider and Virtual Number to Fonoster using the command-line
4. Create a Fonoster Domain using the command line (Assign the Virtual Number to your Domain)
5. Create a Fonoster Agent using the command-line
6. Register to Fonoster using your Agent’s credentials

At that point, you will be able to make calls. I recommend testing this with Zoiper first. But once everything is set, you can use the [wphone](https://github.com/fonoster/wphone) library for calls from the browser.

**Reply by itzmanish** [reply]
I have created a fonoster agent but unable to register that using my mobile sip account settings.

**Reply by psanders** [reply]
Can you show your configuration?

**Reply by itzmanish** [reply]
what configs to show you?

**Reply by psanders** [reply]
It sounds like you want to connect using a Softphone on your cellphone. Is that correct?

**Reply by itzmanish** [reply]
yes correct.

**Reply by psanders** [reply]
Got you. Can you show the parameters on your Softphone?

**Reply by itzmanish** [reply]
port number - 5060
password - *******
username - manish
server - fonoster.sip.twilio.com.fonoster.io

**Reply by psanders** [reply]
Do you have an option that is similar to "outbound proxy"? For example with Zoiper I get "Unauthorized" which means that it is connecting.

<img width="840" alt="Screen Shot 2021-11-23 at 9 43 44 AM" src="https://user-images.githubusercontent.com/539774/143045339-a9dc3671-bad2-4f74-9424-ceeddf73e72a.png">

**Comment by itzmanish** [comment]
Yes there is outbound proxy field

**Reply by itzmanish** [reply]
I added sip.fonoster.io in outbound proxy field but now registration request gets timeout

**Reply by itzmanish** [reply]
It's working fine
